File: Zapp Account/examples/deepseek_test.js
// Node example: prints sample curl and optionally performs request
const key = process.env.DEEPSEEK_API_KEY;
const url = process.env.DEEPSEEK_API_URL || 'https://api.deepseek.example/v1/search';

if (!key) {
  console.error('ERROR: DEEPSEEK_API_KEY is not set.');
  process.exit(1);
}

console.log('DEEPSEEK_API_URL=' + url);
console.log('DEEPSEEK_API_KEY found in environment (hidden)');

const sample = { query: 'test' };
const model = process.env.DEEPSEEK_MODEL;
if (model) sample.model = model;

console.log('\nSample curl command:\n');
console.log(
  `curl -H "Authorization: Bearer $DEEPSEEK_API_KEY" -H "Content-Type: application/json" -X POST "${url}" -d '${JSON.stringify(sample)}'`
);

if (process.env.DEEPSEEK_PERFORM_REQUEST === '1') {
  const fetch = globalThis.fetch || require('node-fetch');
  (async () => {
    const resp = await fetch(url, {
      method: 'POST',
      headers: { Authorization: `Bearer ${key}`, 'Content-Type': 'application/json' },
      body: JSON.stringify(sample),
    });
    console.log('\nHTTP', resp.status);
    try {
      const j = await resp.json();
      console.log(j);
    } catch (e) {
      console.log(await resp.text());
    }
  })();
}

File: Zapp Account/examples/deepseek_test.py
import os
import json
import sys

KEY = os.environ.get("DEEPSEEK_API_KEY")
URL = os.environ.get("DEEPSEEK_API_URL", "https://api.deepseek.example/v1/search")

if not KEY:
    print("ERROR: DEEPSEEK_API_KEY is not set. Set it globally or in this terminal.")
    sys.exit(1)

print(f"DEEPSEEK_API_URL={URL}")
print("DEEPSEEK_API_KEY found in environment (hidden)")

sample_payload = {"query": "test"}
model = os.environ.get("DEEPSEEK_MODEL")
if model:
    sample_payload["model"] = model

print("\nSample curl command to run:\n")
print(
    (
        f"curl -H \"Authorization: Bearer $DEEPSEEK_API_KEY\" -H \"Content-Type: application/json\" "
        f"-X POST \"{URL}\" -d '{json.dumps(sample_payload)}'"
    )
)

if os.environ.get("DEEPSEEK_PERFORM_REQUEST") == "1":
    try:
        import requests
    except Exception:
        print("requests not installed. Install with: pip install requests")
        sys.exit(1)

    headers = {"Authorization": f"Bearer {KEY}", "Content-Type": "application/json"}
    resp = requests.post(URL, headers=headers, json=sample_payload)
    print("\nHTTP", resp.status_code)
    try:
        print(resp.json())
    except Exception:
        print(resp.text)
